mmeds/database.py

```diff
@@ -197,25 +197,33 @@ def format(self, text, header=None):
 
     def execute(self, sql):
         """ Execute the provided sql code """
+        header = None
+        # If the user is not an admin automatically map tables in the query
+        # to their protected views
+        if not self.user == 'root':
+            for table in fig.PROTECTED_TABLES:
+                sql = sql.replace(' ' + table + ' ', ' protected_' + table + ' ')
```

Comment:
As we discussed, I think this is fine but I'd recommend testing with corner cases. Like a query that joins protected and not-protected tables or something like that.

Reply:
Unless you've done that already

Reply:
I have not, I'll add that